bluepysnap/config.py

```diff
@@ -35,29 +35,32 @@ class Config(object):
         https://github.com/AllenInstitute/sonata/blob/master/docs/SONATA_DEVELOPER_GUIDE.md#network_config
     """
 
-    def __init__(self, filepath):
+    def __init__(self, config):
         """Initializes a Config object from a path to the actual config.
 
         Args:
-            filepath (str): Path the SONATA configuration file.
+            config (str/dict): Path to the SONATA configuration file or dict containing the config.
```

Comment:
Actually this is on purpose. Doing it here allow to factorize the `if/else` instead of delegating it to all the consumer class. 
So basically not having the `if/else` here would mean to have exactly the same code in all the consumers: 
```
    if isinstance(config, dict):
         self._config = Config.load_dict().resolve()
    elif isinstance(config, ...):
         ....
```
Which is not optimal imao.